internal/shared/dto/currency/currency.go

```diff
@@ -0,0 +1,10 @@
+package currency
+
+type ResponseDTO struct {
+	Number float64 `json:"number"`
+}
+
+type CachedCurrency struct {
+	UpdateDate string `json:"updateDate"`
+	ResponseDTO
+}
```

Comment:
Are you sure this structure should be shared? You'll use it both as the repository return  value and as the serialized payload in the API response.